### PR TITLE
add audiences fields

### PR DESCRIPTION
**Pull request by elliotCamblor**

### CODE REVIEW

**Review by ajwootto — APPROVED** [review]

**Review by nsalamad — APPROVED** [review]

File: lib/shared/bucketing-assembly-script/assembly/types/configBody.ts [hunk 1]
```diff
@@ -97,7 +99,15 @@ export class ConfigBody {
...
             return new Feature(feature as JSON.Obj)
         })
-
+        const audiences = getJSONObjFromJSON(configJSONObj, 'audiences')
+        this.audiences = new Map<string, Audience>()
+        const audienceKeys = audiences.keys
+        for (let i=0; i < audienceKeys.length; i++) {
+            const audience_id = audienceKeys[i]
+            const aud = audiences.get(audience_id)
+            if (!aud) throw new Error(`Unable to get audience value for id: ${audience_id}`)
```
Comment (ajwootto) [inline lib/shared/bucketing-assembly-script/assembly/types/configBody.ts:108]: how is this possible?

File: lib/shared/bucketing-test-data/src/data/testData.ts [hunk 2]
```diff
@@ -248,6 +248,7 @@ export const variableHashes: ConfigBody['variableHashes'] = {
 export const config: ConfigBody = {
     project,
     environment,
+    audiences: {},
```
Comment (ajwootto) [inline lib/shared/bucketing-test-data/src/data/testData.ts:251]: should you add some test data here or is that a different PR?

File: lib/shared/bucketing-assembly-script/assembly/types/configBody.ts (hunk 1 above)
Comment (elliotCamblor) [reply lib/shared/bucketing-assembly-script/assembly/types/configBody.ts:108]: I don't think it should be, I was just replicating what I saw in the variableHashes code. I can delete it

Comment (elliotCamblor) [reply lib/shared/bucketing-assembly-script/assembly/types/configBody.ts]: Its been removed

File: lib/shared/bucketing-assembly-script/assembly/types/target.ts [hunk 3]
```diff
@@ -131,6 +133,13 @@ export class AudienceFilterOrOperator extends JSON.Value {
             this.values = null
         }
 
+        const _audiencesArr = filter.getArr('_audiences')
```
Comment (ajwootto) [inline lib/shared/bucketing-assembly-script/assembly/types/target.ts:136]: isnt there already a helper for this block? like getArrayFromJSONOptional or something?

Comment (elliotCamblor) [reply lib/shared/bucketing-assembly-script/assembly/types/target.ts:136]: It doesnt look like that particular helper is a thing. I could make it though

Comment (ajwootto) [reply lib/shared/bucketing-assembly-script/assembly/types/target.ts:136]: looks like it would be useful

Comment (elliotCamblor) [reply lib/shared/bucketing-assembly-script/assembly/types/target.ts]: yea, added

**Review by ajwootto — APPROVED** [review]

**Pull request merged**